File: TP3-CPP/Kode Program/TripKuliner.cpp
#pragma once
#include <string>
#include <vector>
#include <iostream>
#include "Trip.cpp"
using namespace std;

//class unutk trip kuliner 
class TripKuliner : public Trip {
private:
    //vector untuk meyimpan daftar
    vector<string> daftar_makanan;
    vector<string> daftar_tempat;

public:
    // Konstruktor TripKuliner, Menginisialisasi data trip umum dari class Trip serta data spesifik kuliner 
    TripKuliner(string id_trip, string nama_trip, double harga, string lokasi, JadwalTrip jadwal,
                vector<string> daftar_makanan, vector<string> daftar_tempat)
        : Trip(id_trip, nama_trip, harga, lokasi, jadwal),
          daftar_makanan(daftar_makanan), daftar_tempat(daftar_tempat) {}
    
    // Override method tampilkan_data() dari class Trip
    // Menambahkan informasi khusus tentang trip kuliner
    void tampilkan_data() const override {
        // Tampilkan data umum trip (dipanggil dari superclass)
        Trip::tampilkan_data();

        // Tampilkan daftar makanan
        cout << "Daftar Makanan: ";
        for (const auto& m : daftar_makanan) cout << m << ", ";

        // Tampilkan daftar tempat kuliner
        cout << "\nDaftar Tempat: ";
        for (const auto& t : daftar_tempat) cout << t << ", ";
        cout << endl;
    }
};

File: TP3-Java/Kode Program/TripKuliner.java
import java.util.List;

public class TripKuliner extends Trip {
    private List<String> daftarMakanan;
    private List<String> daftarTempat;

    // Konstruktor untuk menginisialisasi objek TripKuliner
    public TripKuliner(String idTrip, String namaTrip, double harga, String lokasi,
                       JadwalTrip jadwal, List<String> daftarMakanan, List<String> daftarTempat) {
        // Memanggil konstruktor dari kelas induk (Trip)
        super(idTrip, namaTrip, harga, lokasi, jadwal);
        this.daftarMakanan = daftarMakanan;
        this.daftarTempat = daftarTempat;
    }

    @Override
    public void tampilkanData() {
        // Menampilkan data umum trip dari kelas induk
        super.tampilkanData();

        // Menampilkan daftar makanan
        System.out.print("Daftar Makanan: ");
        for (String m : daftarMakanan) {
            System.out.print(m + ", ");
        }

        // Menampilkan daftar tempat kuliner
        System.out.print("\nDaftar Tempat: ");
        for (String t : daftarTempat) {
            System.out.print(t + ", ");
        }
        System.out.println();
    }
}
